File: app/src/main/java/me/ycdev/android/devtools/sampler/SamplerUtils.java
package me.ycdev.android.devtools.sampler;

import android.os.Environment;

import java.io.File;

import me.ycdev.android.devtools.utils.Constants;

public class SamplerUtils {
    public static File getSamplerFolder() {
        File sdRoot = Environment.getExternalStorageDirectory();
        return new File(sdRoot, Constants.EXTERNAL_STORAGE_PATH_APPS_SAMPLER);
    }

    public static File getFileForSampler(String fileName, boolean mkdir) {
        File appDir = getSamplerFolder();
        if (mkdir) {
            //noinspection ResultOfMethodCallIgnored
            appDir.mkdirs();
        }
        return new File(appDir, fileName);
    }
}

File: app/src/main/java/me/ycdev/android/devtools/sampler/SamplerUtils.kt
package me.ycdev.android.devtools.sampler

import android.content.Context
import me.ycdev.android.devtools.utils.AppConstants
import java.io.File

object SamplerUtils {
    fun getSamplerFolder(context: Context): File {
        val appRoot = context.getExternalFilesDir(null) ?: context.filesDir
        return File(
            appRoot,
            AppConstants.EXTERNAL_STORAGE_PATH_APPS_SAMPLER,
        )
    }

    fun getFileForSampler(
        context: Context,
        fileName: String,
        mkdir: Boolean,
    ): File {
        val appDir = getSamplerFolder(context)
        if (mkdir) {
            appDir.mkdirs()
        }
        return File(appDir, fileName)
    }
}
